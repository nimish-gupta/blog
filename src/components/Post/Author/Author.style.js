import styled from 'styled-components';

import {
	breakpointSm,
	lineHeight,
	marginTop,
	marginBottom,
} from '../../../assets/styled-components/mixins';

export const Author = styled.div`
	border-top: 1px solid ${(props) => props.theme.color.grayBorder};
	max-width: ${(props) => props.theme.layout.postWidth};
	padding-top: 20px;
	${lineHeight(1)}
	${marginTop(1)}
	${marginBottom(1)}
	${breakpointSm`
    margin-left: auto;
    margin-right: auto;
  `}
`;

export const Twitter = styled.a`
	display: block;
	text-decoration: underline;
`;
