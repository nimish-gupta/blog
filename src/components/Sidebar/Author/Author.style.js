/* eslint-disable prefer-template */
import styled from 'styled-components';
import { math } from 'polished';
import { Link } from 'gatsby';

import {
	margin,
	marginBottom,
	lineHeight,
} from '../../../assets/styled-components/mixins';

export const Photo = styled.img`
	display: inline-block;
	margin-bottom: 0;
	border-radius: 50%;
	background-clip: padding-box;
`;

const Title = (comp) => styled[comp]`
	font-size: ${(props) =>
		math(props.theme.typographic.baseFontSize + ' * 1.125')};
	font-weight: 600;
	${lineHeight(1.125)};
	${margin(0.5, 0, 0.5, 0)};
`;

export const TitleLink = styled(Link)`
	color: ${(props) => props.theme.color.base};

	&:hover,
	&:focus {
		color: ${(props) => props.theme.color.base};
	}
`;

export const TitleH2 = Title('h2');
export const TitleH1 = Title('h1');

export const Subtitle = styled.p`
	color: ${(props) => props.theme.color.gray};
	${lineHeight(1)};
	${marginBottom(1)};
`;
