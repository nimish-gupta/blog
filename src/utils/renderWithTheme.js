import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
