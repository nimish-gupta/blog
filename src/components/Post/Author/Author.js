// @flow strict
import React from 'react';

import { getContactHref } from '../../../utils';
import * as Styled from './Author.style';
import { useSiteMetadata } from '../../../hooks';

const Author = () => {
	const { author } = useSiteMetadata();

	return (
		<Styled.Author>
			<p>
				{author.bio}
				<Styled.Twitter
					href={getContactHref('twitter', author.contacts.twitter)}
					rel="noopener noreferrer"
					target="_blank">
					<strong>{author.name}</strong> on Twitter
				</Styled.Twitter>
			</p>
		</Styled.Author>
	);
};

export default Author;
