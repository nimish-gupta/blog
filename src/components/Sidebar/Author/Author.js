// @flow strict
import React from 'react';
import { withPrefix, Link } from 'gatsby';
import * as Styled from './Author.style';

type Props = {
	author: {
		name: string,
		bio: string,
		photo: string,
	},
	isIndex: ?boolean,
};

const Author = ({ author, isIndex }: Props) => (
	<div>
		<Link to="/">
			<Styled.Photo
				src={withPrefix(author.photo)}
				width="75"
				height="75"
				alt={author.name}
			/>
		</Link>

		{isIndex === true ? (
			<Styled.TitleH1>
				<Styled.TitleLink to="/">{author.name}</Styled.TitleLink>
			</Styled.TitleH1>
		) : (
			<Styled.TitleH2>
				<Styled.TitleLink to="/">{author.name}</Styled.TitleLink>
			</Styled.TitleH2>
		)}
		<Styled.Subtitle>{author.bio}</Styled.Subtitle>
	</div>
);

export default Author;
