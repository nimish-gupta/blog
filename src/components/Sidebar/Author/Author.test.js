// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Author from './Author';

describe('Author', () => {
	const props = {
		author: {
			name: 'test',
			photo: '/photo.jpeg',
			bio: 'test',
		},
		isIndex: false,
	};

	it('renders correctly', () => {
		const tree = renderWithTheme(<Author {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
