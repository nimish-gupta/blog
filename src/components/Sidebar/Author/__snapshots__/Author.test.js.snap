// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="test"
      className="sc-AxjAm jmjnph"
      height="75"
      src="/photo.jpeg"
      width="75"
    />
  </a>
  <h2
    className="sc-AxiKw ewVnCq"
  >
    <a
      className="sc-AxirZ cvZzWo"
      href="/"
    >
      test
    </a>
  </h2>
  <p
    className="sc-AxhUy dCfmEK"
  >
    test
  </p>
</div>
`;
